### ISSUE TITLE
Cannot connect elk-traffic-monitor-api to elastic search with HTTPS and credentials

### ISSUE THREAD

**Issue by jojo221119** [Axway-API-Management-Plus/apigateway-openlogging-elk#27]
Our productive Elasticsearch instance is served via HTTPS and requires authentication.
The certificate used for the elastic endpoint is signed by an internal CA.

I changed the configuration in "elasticsearch.default.js" and run a docker build. But still the container fails as soon as an API endpoint which connects to Elasticsearch is called.

I changed the configuration in the following way:

```
node: (process.env.ELASTIC_NODE) ? process.env.ELASTIC_NODE : 'http://elasticsearch1:9200', 
auth: {
	/* Use an API-Key
	apiKey: process.env.ELASTIC_API_KEY
		or username / password based
	*/
	username: "user",
	// process.env.ELASTIC_USERNAME, 
	username: "password"
	// process.env.ELASTIC_PASSWORD

}, 
ssl: {
	ca: [
		`-----BEGIN CERTIFICATE-----
		-----END CERTIFICATE-----`,
		`-----BEGIN CERTIFICATE-----
		-----END CERTIFICATE-----`
	]
},
// The name to identify the client instance in the events.
name: 'elk-traffic-monitor-api'
```

The Error message is:

```
elk-traffic-monitor-api_1  | 1599723366498  [request-id: 0f7ecb38-c2a9-4f05-9892-fd432cd62d5c]   search (elasticsearch.1) Error returned from Elastic-Search: {"name":"ConnectionError","meta":{"body":null,"statusCode":null,"headers":null,"warnings":null,"meta":{"context":null,"request":{"params":{"method":"POST","path":"/apigw-traffic-details*/_search","body":"{\"query\":{\"bool\":{\"must\":[{\"range\":{\"@timestamp\":{\"gt\":1599636966455}}},{\"term\":{\"processInfo.serviceId\":\"instance-4\"}}]}}}","querystring":"size=1000&sort=%40timestamp%3Adesc","headers":{"User-Agent":"elasticsearch-js/7.6.1 (linux 3.10.0-1127.18.2.el7.x86_64-x64; Node.js v12.18.3)","Content-Type":"application/json","Content-Length":"123"},"timeout":60000},"options":{"ignore":[404],"maxRetries":3,"warnings":null},"id":1},"name":"elasticsearch-js","connection":{"url":"https://7f4d70cee3af479e96fe97d575819772.rb-elasticsearch.de.bosch.com:9243/","id":"https://7f4d70cee3af479e96fe97d575819772.rb-elasticsearch.de.bosch.com:9243/","headers":{},"deadCount":0,"resurrectTimeout":0,"_openRequests":0,"status":"alive","roles":{"master":true,"data":true,"ingest":true,"ml":false}},"attempts":3,"aborted":false}}}
elk-traffic-monitor-api_1  | Uncaught Exception TypeError: Cannot read property 'error' of undefined
elk-traffic-monitor-api_1  |     at /app/node_modules/@axway-api-builder-ext/api-builder-plugin-fn-elasticsearch/src/actions/search.js:85:40
elk-traffic-monitor-api_1  |     at f (/app/node_modules/once/once.js:25:25)
elk-traffic-monitor-api_1  |     at onResponse (/app/node_modules/@elastic/elasticsearch/lib/Transport.js:221:16)
elk-traffic-monitor-api_1  |     at ClientRequest.<anonymous> (/app/node_modules/@elastic/elasticsearch/lib/Connection.js:98:9)
elk-traffic-monitor-api_1  |     at ClientRequest.emit (events.js:315:20)
elk-traffic-monitor-api_1  |     at TLSSocket.socketErrorListener (_http_client.js:426:9)
elk-traffic-monitor-api_1  |     at TLSSocket.emit (events.js:315:20)
elk-traffic-monitor-api_1  |     at emitErrorNT (internal/streams/destroy.js:92:8)
elk-traffic-monitor-api_1  |     at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
elk-traffic-monitor-api_1  |     at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

Could you provide information whether this is supported and how to investigate this further?

**Comment by cwiechmann** [comment]
Hi @jojo221119, that is actually not yet supported.

**Comment by jojo221119** [comment]
Hi @cwiechmann,

I was able to fix the certificate topic by mounting the CAs in a Volume and configuring the env variable "NODE_EXTRA_CA_CERTS"

The only missing part is the possibility to provide user name and password for elastic search.
It works if I add the information hard coded but as soon as I try to configure it via env variables it stops working.

Do you have an idea?

**Comment by cwiechmann** [comment]
Hi @jojo221119, yes, this will be part of the initial release. I have changed the complete setup to use HTTPS internally. Also Elasticsearch now supports Username/Password based authentication. 
Only missing part is the API-Builder endpoint switched to HTTPS, which requires a certificate validation by Logstash.

**Comment by cwiechmann** [comment]
Hello @jojo221119, I have released a new release package: 1.0.0-RC1 which supports username/password based authentication to Elasticsearch. 
You no longer clone the Github-Project, instead download the release package from here:
https://github.com/Axway-API-Management-Plus/apigateway-openlogging-elk/releases

The provided .env file actually contains most of the documentation and I'm working on to improve the readme file. Nevertheless, I propose to download this release and study the .env file for now.

This release also already contains a number of other improvements. I'm happy to get you feedback.
